### PR TITLE
Travis Stack integration

### PR DESCRIPTION
**Pull request by psibi**
This only does stack integration. Automatic deployment of pages will be done as a next step on top of this.

### CODE REVIEW

**Comment by psibi** [comment]
It took more time than I expected as I was in the middle of switching job and moving to Bangalore. I will try to do a PR soon on automatic deployment of pages. But free feel to do it before me.

File: .travis.yml [hunk 1]
```diff
@@ -7,37 +7,48 @@ notifications:
...
-  - cabal configure
-  - cabal build  # build the site.hs
-  - ./dist/build/website/website build  # build the website contents
-  - ./dist/build/website/website check  # check for broken links.
-  
+- |
+  set -ex
+  case "$BUILD" in
+    stack)
+      stack --no-terminal $ARGS test --bench --no-run-benchmarks --haddock --no-haddock-deps
```
Comment (piyush-kurur) [inline .travis.yml]: Do we need --haddock ? this is just the executable that matters. Also we do not have tests here. So may be `stack build` is sufficient

File: .travis.yml [hunk 2]
```diff
@@ -7,37 +7,48 @@ notifications:
...
-    - $HOME/.cabal/packages
+    - $HOME/.ghc   
+    - $HOME/.cabal
+    - $HOME/.stack
 
+before_install:
+# Download and unpack the stack executable
+- mkdir -p ~/.local/bin
+- export PATH=$HOME/.local/bin:$PATH
+- travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
```
Comment (piyush-kurur) [inline .travis.yml:39]: If fpcomplete repos are available then we can also use addons toget the stack executable.

**Comment by piyush-kurur** [comment]
I will not be able to merge this immediately will do it after 25th (need to get my slides ready for haskell2016).

**Comment by psibi** [comment]
@piyush-kurur I will address your comments in a day or so. 

> need to get my slides ready for haskell2016

Just seeing about the symposium and wow that seems exciting! All the best.

**Comment by piyush-kurur** [comment]
I will just merge you changes. But may be you can work on the comments that I said as part of your next pull request.

**Comment by psibi** [comment]
Thanks.

**Pull request merged**